Source/WebCore/Modules/model-element/HTMLModelElement.cpp

```diff
@@ -693,6 +698,17 @@ void HTMLModelElement::setCurrentTime(double currentTime)
         m_modelPlayer->setCurrentTime(Seconds(currentTime), [&]() { });
 }
 
+bool HTMLModelElement::hasPortal() const
+{
+    return !hasAttributeWithoutSynchronization(HTMLNames::noportalAttr);
```

Comment:
This should probably also check `document().settings().nonPortaledModelEnabled()`?
e.g. `return document().settings().nonPortaledModelEnabled() && !hasAttributeWithoutSynchronization(HTMLNames::noportalAttr)`.

Reply:
Should we move the check here and get rid of the one in `attributeChanged`?

Reply:
Modified to `return !document().settings().nonPortaledModelEnabled() || hasAttributeWithoutSynchronization(HTMLNames::noportalAttr)` since this should default to true.